spinn_utilities/citation/citation_and_doi_updater.py

```diff
@@ -0,0 +1,398 @@
+                # are not in the avoids set
+                avoid = False
+                if potential_zip_file in avoids:
+                    avoid = True
+                for directory_name in directories:
+                    if directory_name in avoids:
+                        avoid = True
+                for directory_name in directory_path.split(os.sep):
+                    if directory_name in avoids:
+                        avoid = True
+
+                # if safe to zip, zip
+                if not avoid:
+                    module_zip_file.write(
+                        os.path.join(directory_path, potential_zip_file))
+
+    @staticmethod
+    def _fill_in_data(doi_title, doi_description, yaml_file):
+        """ adds in data to the zenodo metadata
+
+        :param doi_title: the title of the doi
+        :type doi_title: str
+        :param doi_description: the description of the doi
+        :type doi_description: str
+        :param yaml_file: the citation file once read into the system
+        :type yaml_file: dict
+        :return: dict containing zenodo metadata
+        :rtype: dict
+        """
+        data = dict()
+        data[ZENODO_METADATA] = dict()
+
+        # add basic meta data
+        data[ZENODO_METADATA][ZENODO_METADATA_TITLE] = doi_title
+        data[ZENODO_METADATA][ZENODO_METATDATA_DESC] = doi_description
+        data[ZENODO_METADATA][ZENODO_METADATA_CREATORS] = list()
+
+        # get author data from the citation file
+        for author in yaml_file[CITATION_AUTHORS_TYPE]:
+            author_data = dict()
+            author_data[ZENODO_AUTHOR_NAME] = (
+                author[CITATION_AUTHOR_SURNAME] + ", " +
+                author[CITATION_AUTHOR_FIRST_NAME] + ", ")
+            if AUTHOR_AFFILIATION in author:
+                author_data[AUTHOR_AFFILIATION] = author[AUTHOR_AFFILIATION]
+            if AUTHOR_ORCID in author:
+                author_data[AUTHOR_ORCID] = author[AUTHOR_ORCID]
+            data[ZENODO_METADATA][ZENODO_METADATA_CREATORS].append(author_data)
+        return data
+
+    @staticmethod
+    def convert_text_date_to_date(
+            version_month, version_year, version_day):
+        """ converts the 3 components of a date into a CFF date
+
+        :param version_month: version month, in text form
+        :type version_month: text or int
+        :param version_year: version year
+        :type version_year: int
+        :param version_day: version day of month
+        :type version_day: int
+        :return: the string repr for the cff file
+        """
+        return "{}-{}-{}".format(
+            version_year,
+            CitationUpdaterAndDoiGenerator.convert_month_name_to_number(
+                version_month),
+            version_day)
+
+    @staticmethod
+    def convert_month_name_to_number(version_month):
+        """ converts a python month in text form to a number form
+        YUCK> there must be a better way than this
```

Comment:
You could use:
```
from time import strptime
try:
    return strptime(version_month, "%B").tm_mon
except ValueError:
    return strptime(version_month, "%b").tm_mon
```

Note that this might result in a ValueError if the month is not correctly specified i.e. if the month was misspelled.  It does cope with abbreviated and full month names though.